### PR TITLE
fix(tabs): disable without using `pointer-events`

### LINKED ISSUE #6755: tooltip on md-tab not working
#### Bug, feature request, or proposal:
bug

#### What is the expected behavior?
tooltips should be shown on md-tabs

#### What is the current behavior?
tooltips are not shown

#### What are the steps to reproduce?
see plunkr: http://plnkr.co/edit/ij8DMEKht4QTvQt1ANLN?p=preview

- _tab 2 and tab 3 have both tooltips which are not shown_
- _general tooltips work (see on the bottom of the plunkr a working tooltip)_

#### What is the use-case or motivation for changing an existing behavior?
Providing information about tabs would be nice, especially when tabs are set as `disabled`, a reason why it is disabled should be somehow shown to the customer.
There is already a discussion about that on angularjs and old material, so we don't have to discuss the reasons again: https://github.com/angular/material/issues/1667

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
material 1.0.0 beta 10

#### Is there anything else we should know?
if it is intended, what would be a good way to make the user know that this tab is disabled because of reasons (except doing nasty stuff like `<md-tab label="tab2 (not enabled feature)">...</md-tab>`

### PR DESCRIPTION
**Pull request by josephperrott**
Fixes #6755

### CODE REVIEW

File: src/lib/tabs/_tabs-common.scss [hunk 1]
```diff
@@ -24,7 +24,10 @@ $mat-tab-animation-duration: 500ms !default;
 
   &.mat-tab-disabled {
     cursor: default;
-    pointer-events: none;
+
+    &:focus {
```
Comment (crisbeto) [inline src/lib/tabs/_tabs-common.scss]: Fading out the element only on focus seems a little weird. Wouldn't it be better if we prevented focus in the first place? That's what the `pointer-events` approach was doing.

File: src/lib/tabs/tab-group.html [hunk 2]
```diff
@@ -11,8 +11,8 @@
        [attr.aria-selected]="selectedIndex == i"
        [class.mat-tab-label-active]="selectedIndex == i"
        [disabled]="tab.disabled"
-       [matRippleDisabled]="disableRipple"
-       (click)="tabHeader.focusIndex = selectedIndex = i">
+       [matRippleDisabled]="tab.disabled || disableRipple"
+       (click)="!tab.disabled && tabHeader.focusIndex = selectedIndex = i">
```
Comment (crisbeto) [inline src/lib/tabs/tab-group.html]: This expression looks a bit weird. Considering moving it into a function?

File: src/lib/tabs/_tabs-common.scss (hunk 1 above)
Comment (josephperrott) [reply src/lib/tabs/_tabs-common.scss]: By disabling `pointer-events` entirely things like tooltips are prevented from being attached to a `<mat-tab-label>`

**Comment by josephperrott** [comment]
@crisbeto Whenever you have a change, please take another look.

I realized that the issue we were running into was that we were adding `tabindex=-1` to all of the tabs that were not active in an attempt to prevent focusing on them.  However this meant that they could still be focused by mouse.  Instead we will use `[attr.tabIndex]="selectedIndex == i ? 0 : null"` as when `null` is the value, `tabindex` is removed from the element entirely.

File: src/lib/tabs/tab-group.ts [hunk 3]
```diff
@@ -290,4 +291,12 @@ export class MatTabGroup extends _MatTabGroupMixinBase implements AfterContentIn
     this._tabBodyWrapperHeight = this._tabBodyWrapper.nativeElement.clientHeight;
     this._renderer.setStyle(this._tabBodyWrapper.nativeElement, 'height', '');
   }
+
+  /** Handle click events, setting new selected index if appropriate. */
+  _handleClick(tab: MatTab, tabHeader: MatTabHeader, idx: number) {
+    if (!tab.disabled) {
+      tabHeader.focusIndex = idx;
+      this.selectedIndex = idx;
+    }
```
Comment (crisbeto) [inline src/lib/tabs/tab-group.ts]: Optionally this could be reduced to: `this.selectedIndex = tabHeader.focusIndex = idx`.

**Comment by kara** [comment]
@josephperrott Check out CI failures?

**Comment by josephperrott** [comment]
@kara CI failures are fixed.

**Pull request merged**